### PR TITLE
PMM-5291 upgrade clickhouse

### PR DESCRIPTION
**Pull request by nikita-b**
https://github.com/Percona-Lab/pmm-submodules/pull/1876

### CODE REVIEW

File: ansible/playbook/tasks/roles/clickhouse/tasks/main.yml [hunk 1]
```diff
@@ -0,0 +1,76 @@
+---
+- name: Remove old clickhouse packages
+  yum:
+    state: absent
+    name:
+      - percona-clickhouse-client
+      - percona-clickhouse-server
+      - percona-clickhouse-common-static
+      - percona-clickhouse-server-common
```
Comment (BupycHuk) [inline ansible/playbook/tasks/roles/clickhouse/tasks/main.yml:9]: please ensure that old data won't be deleted during upgrade.

**Review by BupycHuk — APPROVED** [review]

File: ansible/playbook/tasks/roles/clickhouse/files/config.xml [hunk 2]
```diff
@@ -0,0 +1,1155 @@
...
+        </levels>
+        -->
+
+        <!-- Per level overrides:
+
+        For example to suppress logging of the RBAC for default user you can use:
+        (But please note that the logger name maybe changed from version to version, even after minor upgrade)
+        -->
+        <!--
+        <levels>
```
Comment (atymchuk) [inline ansible/playbook/tasks/roles/clickhouse/files/config.xml:49]: In essence, the vendor suggests these two contexts be off, i.e. no logs. Do you believe that we still need logs for them?

File: ansible/playbook/tasks/roles/clickhouse/files/config.xml [hunk 3]
```diff
@@ -0,0 +1,1155 @@
...
+    <send_crash_reports>
+        <!-- Changing <enabled> to true allows sending crash reports to -->
+        <!-- the ClickHouse core developers team via Sentry https://sentry.io -->
+        <!-- Doing so at least in pre-production environments is highly appreciated -->
+        <enabled>false</enabled>
+        <!-- Change <anonymize> to true if you don't feel comfortable attaching the server hostname to the crash report -->
+        <anonymize>false</anonymize>
+        <!-- Default endpoint should be changed to different Sentry DSN only if you have -->
+        <!-- some in-house engineers or hired consultants who're going to debug ClickHouse issues for you -->
+        <endpoint>https://6f33034cfe684dd7a3ab9875e57b1c8d@o388870.ingest.sentry.io/5226277</endpoint>
```
Comment (atymchuk) [inline ansible/playbook/tasks/roles/clickhouse/files/config.xml:1150]: Should we send the crash reports there? We may need to ask @rnovikovP  to help us decide.

File: ansible/playbook/tasks/roles/clickhouse/tasks/main.yml (hunk 1 above)
Comment (atymchuk) [reply ansible/playbook/tasks/roles/clickhouse/tasks/main.yml:16]: @nikita-b this removal should not delete data directories, right?

File: ansible/playbook/tasks/roles/clickhouse/tasks/main.yml [hunk 4]
```diff
@@ -0,0 +1,84 @@
...
+    enabled: no
+
+- name: Create clickhouse data directory
+  file:
+    path: "/srv/clickhouse"
+    state: directory
+    owner: root
+    group: pmm
+
+- name: Ensure clickhouse repo GPG key imported
```
Comment (atymchuk) [inline ansible/playbook/tasks/roles/clickhouse/tasks/main.yml:32]:
```suggestion
- name: Import clickhouse repo GPG key
```

**Review by atymchuk — CHANGES_REQUESTED** [review]

File: ansible/playbook/tasks/roles/clickhouse/files/config.xml (hunk 3 above)
Comment (rnovikovP) [reply ansible/playbook/tasks/roles/clickhouse/files/config.xml:1150]: as we are not controlling the "telemetry" status in the CH config file for now, but let PMM users "not being tracked," I would say "turn all related to telemetry and reporting  features off" for now (if this is not adding an additional complication to maintainability of the project )

Comment (atymchuk) [reply ansible/playbook/tasks/roles/clickhouse/files/config.xml:1150]: > as we are not controlling the "telemetry" status in the CH config file for now, but let PMM users "not being tracked," I would say "turn all related to telemetry and reporting features off" for now (if this is not adding an additional complication to maintainability of the project )

I think it should be simple to turn it off.

File: ansible/playbook/tasks/roles/clickhouse/tasks/main.yml (hunk 1 above)
Comment (nikita-b) [reply ansible/playbook/tasks/roles/clickhouse/tasks/main.yml:16]: @BupycHuk @atymchuk It's only remove binaries and packages metadata. I tested the update and it works fine.

File: ansible/playbook/tasks/roles/clickhouse/files/config.xml (hunk 3 above)
Comment (nikita-b) [reply ansible/playbook/tasks/roles/clickhouse/files/config.xml:1150]: It already was turned off. We have `disable` slightly above.

I'll remove `endpoint` option and it'll more transparent.

File: ansible/playbook/tasks/roles/clickhouse/files/config.xml (hunk 2 above)
Comment (nikita-b) [reply ansible/playbook/tasks/roles/clickhouse/files/config.xml:49]: It's just an example from config. I checked and we almost don't have logs from these loggers.

**Review by atymchuk — APPROVED** [review]

**Review by atymchuk — APPROVED** [review]

**Review by BupycHuk — DISMISSED** [review]
We changed log level from trace to info, is it intentional?

**Comment by nikita-b** [comment]
> We changed log level from trace to info, is it intentional?

Yes, we had a lof of Clickhouse logs and we had a separate task to reduce the amount of logs.

**Review by atymchuk — APPROVED** [review]

**Comment by atymchuk** [comment]
> > We changed log level from trace to info, is it intentional?
> 
> Yes, we had a lof of Clickhouse logs and we had a separate task to reduce the amount of logs.

@nikita-b It would be great to say a few words about the less verbose logs in the next Release Notes. It's a clear win if we think how spammy the logs were before.

**Pull request merged**